src/lightning/pytorch/strategies/deepspeed.py

```diff
@@ -893,6 +890,9 @@ def register_strategies(cls, strategy_registry: _StrategyRegistry) -> None:
             offload_optimizer_device="nvme",
         )
 
+    # TODO: discuss should we keep overriding this?
     def batch_to_device(self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0) -> Any:
```

Comment:
Arguments for removing the override: 
- Consistent with other strategies who convert the dtype in the `*_step()` methods
- Otherwise we would be calling `precision_plugin.convert_input(batch)` twice redundantly
- Even if the user overrides the `transfer_batch_to_device` hook and changes the dtype, the dtype would be changed back to what the precision plugin wants before passing it into the LM-step methods.
- The previous behavior could be considered a bug, because the docs of the `transfer_batch_to_device` say:

       

	>  The returned batch is of the same type as the input batch, just
	>         having all tensors on the correct device.
	and
	
	> This hook should only transfer the data and not modify it


Arguments for keeping it:
- Could be considered a breaking change for deepspeed users who expected the dtype to be already converted in `transfer_batch_to_device`

Reply:
Will this be follow-up work?

Reply:
I think we should make a decision here. Waiting for @carmocca's thoughts :)

Reply:
Yes, the batch gets first moved to the device, then passed to the Strategy.training_step methods where the dtype conversion happens (if necessary).

Reply:
We could also do it the other way around, first converting to dtype, then moving to device. This would then be a change that affects all other strategies. Note that in Fabric we do the casting also after device transfer, because the user is generally responsible for making sure the data is on the device before passing it into the model.

Reply:
I'm keeping it for now, not to regress on memory and since there is an explicit test case (thanks for pointing to the PR). I'm following up with a PR that standardizes this.